Kill the comments feature.

punchagan opened the pull request:
Closes #73.

sursh commented:
YAAAASSSS

punchagan commented:
Instead of just killing the link to discussion, I added a link to the zulip discussion, now.

sursh commented:
Cool!

Implementation-wise, this doesn't seem like the safest way to link to the
discussion. What do you think about using the response from the Zulip API
to grab the thread id? This would also allow us to keep track of the Zulip
comment count :) that could be surfaced in the UI and would be really
helpful at identifying the most popular blog posts for the email digest!

-S

On Tue, Mar 10, 2015 at 5:14 PM, Puneeth Chaganti <notifications@github.com>
wrote:

> Instead of just killing the link to discussion, I added a link to the
> zulip discussion, now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sursh/blaggregator/pull/80#issuecomment-78149830>.
>



-- 
-Sasha

punchagan commented:
I agree that the implementation doesn't seem the safest, but I went ahead with it since Zulip is no longer actively developed and the code worked for a bunch of blog posts that I tested it with.  

But, I like your idea! I'll poke around with Zulip API and see what I can come up with.

sursh commented:
This might be helpful! https://gist.github.com/rsms/206437566a81bb076c02

On Tue, Mar 10, 2015 at 6:31 PM, Puneeth Chaganti <notifications@github.com>
wrote:

> Closed #80 <https://github.com/sursh/blaggregator/pull/80>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sursh/blaggregator/pull/80#event-249723773>.
>



-- 
-Sasha

punchagan commented:
It looks like this will take me a while to figure out and work on. I'll pull out the commit that adds the zulip link from this PR, and add the discussed changes in a separate PR.

punchagan commented:
It doesn't look like there's a way to get the URL for a topic/subject from the API. I've added back the commit that adds the Zulip link. The code currently does exactly what is done in Zulip's code to generate the URLs.

lfranchi commented:
I can confirm that this is the "proper" way to generate a url to a stream/topic pair on Zulip :)

punchagan commented:
I rebased this branch onto the `feed-with-token` branch and added a link to the zulip discussion in the feed too. 

I think, we can go ahead with the current implementation of generating the Zulip urls as of now, and come up with something nicer, if and when something breaks and we get any bug reports. If the feature is not used much, we can decide to kill it, instead. There is a possibility, now that Zulip is open source, of adding features to Zulip if we really need something!

sursh commented on home/feedergrabber27.py:
```diff
@@ -103,9 +102,15 @@ def feedergrabber(url=None):
             post_date = datetime.datetime.now()
+
+        # Post content
+        content = getattr(i, 'summary', '')
```
what's this bit for?

sursh commented on blaggregator/settings.py:
```diff
@@ -8,7 +8,6 @@
 
```
Thanks for clearing out the cruft as you go!

sursh commented on home/static/css/mystyles.css:
```diff
@@ -46,37 +46,6 @@ a.muted {
   color: #85ACC0;
 }
 
-/* Comments */
```
I am just LOVING all this red!

sursh commented on home/static/css/mystyles.css:
```diff
@@ -137,18 +106,18 @@ div.bar-container {
-}
\ No newline at end of file
+}
+.feed-icon {
```
This style isn't getting applied for some reason that isn't obvious to me (at least in Chrome). The logo is hugging the top of the screen. 

![screen shot 2015-11-09 at 10 08 45 pm](https://cloud.githubusercontent.com/assets/719590/11053377/8055df26-872e-11e5-97c3-2be062cb9c6d.png)

punchagan replied:
Looks like the CSS didn't get deployed for some reason, and I'm [unable to re-deploy](https://devcenter.heroku.com/articles/cedar-14-migration) stuff onto staging. The CSS changes do appear on localhost.

sursh commented on home/templates/home/base.html:
```diff
@@ -39,8 +39,15 @@
+
+                {% if user.is_authenticated %}
+                  <a href="{% url 'feed' %}?token={{user.hacker.token}}" target="_blank">
+                    <img class="feed-icon pull-right" src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/Rss-feed.svg/24px-Rss-feed.svg.png"/>
```
:fire: :link:

sursh commented on home/templates/home/item.html:
```diff
@@ -22,36 +22,4 @@ <h4 class="media-heading"><a href="view">{{ post.title }}</a></h4>
     </div>
```
Do we need this view at all anymore? I don't think you can actually get to it anymore.

punchagan replied:
Yep, we don't need it anymore! Its gone and in comes more red! :)

sursh commented on home/templates/home/new.html:
```diff
@@ -21,15 +21,14 @@
+            </h4>
+            <div>
+              <a href="{{ post.title|zulip_url:post.stream }}" target="_blank" class="listing-comments">
+                Join the discussion (on Zulip)
```
I'm not too attached to the old wording "Join the discussion". Happy to simplify this down to "Comments on Zulip" or "Discuss on Zulip"

punchagan replied:
Done.

sursh commented on home/templates/home/profile.html:
```diff
@@ -22,23 +22,47 @@
+          <a href="{% url 'feed' %}?token={{hacker.token}}">RSS feed</a>.  Please keep
+          your feed URL private.  In case you accidentally share the URL, you
+          can reset it using the button below.
+        </p>
```
Love it, this is very clear messaging.

sursh commented on home/templatetags/customtags.py:
```diff
@@ -19,3 +21,14 @@ def pagination(pages, page):
+    # So Zulip hides the URI-encoding by replacing '%' with '.'
+    replace = lambda x: quote(x, safe='').replace('.', '%2E').replace('%', '.')
+    hash_path = 'narrow/stream/%s/topic/%s' % (replace(stream), replace(title))
+
```
Might be worth documenting for posterity in the comments in the code (as well as on this PR) that this is the Right Way to do URL construction.

punchagan replied:
Done.

sursh commented on home/views.py:
```diff
@@ -265,47 +254,49 @@ def updated_avatar(request, user_id):
 
-    post = get_post_info(slug)
+    The model is passed as an argument to allow using this function in data
+    migrations.
```
Sorry, where is the model passed? I'm a little confused by this comment.

punchagan replied:
This is an outdated docstring. The code has been duplicated in the migration to allow backward/forward migrations without depending on this function to exist in the app's code.

sursh commented:
Clicking on the RSS logo isn't working for me. I'm getting the following error: 

```

Template error:
In template /app/home/templates/home/feed_item.tmpl, error at line 8
   ’
   1 : {% load customtags %}


   2 : 


   3 : {{ obj.content|safe }}


   4 : 


   5 : <p>


   6 :   This post was originally published <a href="{{ obj.url }}"


   7 :   target="_blank">here</a> and is brought to you by <a href="{{ site.domain }}"


   8 :   >Blaggregator</a>. You can <a href=" {{ obj.title|zulip_url:obj.blog.stream }} " target="_blank"> join


   9 :   the discussion </a> on Zulip!.


   10 : </p>


   11 : 

Traceback:
File "/app/.heroku/python/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  115.                         response = callback(request, *callback_args, **callback_kwargs)
File "/app/home/views.py" in feed
  265.     return LatestEntriesFeed()(request)
File "/app/.heroku/python/lib/python2.7/site-packages/django/contrib/syndication/views.py" in __call__
  44.         feedgen = self.get_feed(obj, request)
File "/app/.heroku/python/lib/python2.7/site-packages/django/contrib/syndication/views.py" in get_feed
  154.                 description = description_tmp.render(RequestContext(request, {'obj': item, 'site': current_site}))
File "/app/.heroku/python/lib/python2.7/site-packages/django/template/base.py" in render
  140.             return self._render(context)
File "/app/.heroku/python/lib/python2.7/site-packages/django/template/base.py" in _render
  134.         return self.nodelist.render(context)
File "/app/.heroku/python/lib/python2.7/site-packages/django/template/base.py" in render
  830.                 bit = self.render_node(node, context)
File "/app/.heroku/python/lib/python2.7/site-packages/django/template/debug.py" in render_node
  74.             return node.render(context)
File "/app/.heroku/python/lib/python2.7/site-packages/django/template/debug.py" in render
  84.             output = self.filter_expression.resolve(context)
File "/app/.heroku/python/lib/python2.7/site-packages/django/template/base.py" in resolve
  606.                 new_obj = func(obj, *arg_vals)
File "/app/home/templatetags/customtags.py" in zulip_url
  32.     hash_path = 'narrow/stream/%s/topic/%s' % (replace(stream), replace(title))
File "/app/home/templatetags/customtags.py" in <lambda>
  31.     replace = lambda x: quote(x, safe='').replace('.', '%2E').replace('%', '.')
File "/app/.heroku/python/lib/python2.7/urllib.py" in quote
  1294.     return ''.join(map(quoter, s))

Exception Type: KeyError at /feed/
Exception Value: u'\u2019'

```

punchagan commented:
Thanks for the review, @sursh. I'll address your comments in a day or two. I'm currently traveling.

punchagan commented:
There was a missing encode to utf-8 call, which was causing the crash reported above. It should now be fixed.

punchagan commented:
I think I've addressed all your comments, but I'm unable to [deploy to staging](https://devcenter.heroku.com/articles/cedar-14-migration). I'll get to it in a couple of days or so, if you don't already get to it. Thanks!

punchagan commented:
I've upgraded the stack on staging to cedar-14 and redeployed this branch.

FiloSottile commented:
Howdy! Status of this? I can't wait to read Blaggregator in my RSS feed :)

punchagan commented:
@FiloSottile Sorry to keep you waiting so long!  I'm eagerly waiting for this to be merged and deployed too! @sursh is pretty busy for the next couple of weeks or so. Hopefully, we should merge it in soon and give ourselves a good new year gift! :fireworks:

punchagan commented:
@FiloSottile Note that currently the feed doesn't have the full content of the posts, though, it should be an easy change if desired.

FiloSottile commented:
:) Full content would be nice (even if the client I use can extract with
:Readability). Let me know if you want me to open an issue to
:discuss/track it.

punchagan commented:
Well actually, I'm wrong. A 6-month-younger me had implemented the code to save post content and display it in the feed.  I was looking at feeds generated from the current db dump and assumed that the code to fix crawling wasn't implemented. But, this confusion got me to learn about RSS readers integrated with Readability! :+1:

FiloSottile commented:
I hate to be a nag, but is there any progress on deploying this? I'd be happy to help if I can!

jvns commented:
I'd also love to have this feature :heart:

punchagan commented:
@FiloSottile @jvns This is finally available.  Sorry that it took us so long to get this out of the door, but you can finally use this.  You can get the feed URL from the header on the [home page of Blaggreagtor](https://blaggregator.recurse.com)
